packit_service/worker/handlers/bodhi.py

```diff
@@ -63,15 +78,92 @@ def __init__(
         )
 
         # lazy properties
-        self._koji_build_event: Optional[KojiBuildEvent] = None
+        self._koji_build_event_data: Optional[KojiBuildEventWrapper] = None
+        self._packit_api: Optional[PackitAPI] = None
+        self._local_project: Optional[LocalProject] = None
+
+    def _build_from_event_dict(self) -> KojiBuildEventWrapper:
+        koji_build_event = KojiBuildEvent.from_event_dict(self.data.event_dict)
+        return KojiBuildEventWrapper(
+            build_id=koji_build_event.build_id,
+            state=koji_build_event.state,
+            dist_git_branch=koji_build_event.git_ref,
+            nvr=koji_build_event.nvr,
+        )
+
+    def _build_from_koji_api(self) -> KojiBuildEventWrapper:
+        dist_git_branch = self.project.get_pr(self.data.pr_id).target_branch
+        nvr = DistGit.get_latest_build_in_tag(
+            downstream_package_name=self.project.repo, dist_git_branch=dist_git_branch
+        )
+
+        session = ClientSession(baseurl=KOJI_BASEURL)
+        build = session.getBuild(buildInfo=nvr)
```

Comment:
I think that we could refactor the packit `get_latest_build_in_tag` method to return the whole build dictionary and not only nvr [here](https://github.com/packit/packit/blob/0d8c2b41e22e4477dfa7550723df0ab0cfd0a38b/packit/distgit.py#L449) and then we would not need to call the Koji API here the second time